### PR TITLE
iox-#1345 Minimal logger re-implementation

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [ ] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [ ] Tests follow the [best practice for testing][testing]
1. [ ] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [ ] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [ ] Commits messages are according to this [guideline][commit-guidelines]
    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
    - [ ] Commit messages are signed (`git commit -s`)
    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [ ] Relevant issues are linked
1. [ ] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] All touched (C/C++) source code files are added to `./clang-tidy-diff-scans.txt`
1. [ ] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] Unit tests have been written for new behavior
- [ ] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [ ] All touched (C/C++) source code files have been added to `./clang-tidy-diff-scans.txt`
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Relates to #1345

### CODE REVIEW

File: tools/scripts/used-headers.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ cstdlib
 cstring
 ctime
 fstream
+functional
```
Comment (mossmaurice) [inline tools/scripts/used-headers.txt:42]: @elBoberido Wouldn't it be possible to use `cxx::function` or maybe even `cxx::function_ref`?

Comment (elBoberido) [reply tools/scripts/used-headers.txt:42]: I have to look why exactly this is used. I just copied my proof of concept to check if it builds and will now remove everything that is not needed

Comment (elBoberido) [reply tools/scripts/used-headers.txt]: @mossmaurice removed

**Comment by elBoberido** [comment]
Closed if favor of https://github.com/eclipse-iceoryx/iceoryx/pull/1569 to be able to create stacked PRs

**Pull request closed without merging**